src/BuildToolchains.jl

```diff
@@ -234,7 +234,7 @@ function generate_toolchain_files!(platform::AbstractPlatform, envs::Dict{String
         symlink_if_exists(target, link) = ispath(joinpath(dir, target)) && symlink(target, link)
 
         # On FreeBSD and MacOS we actually want to default to clang, otherwise gcc
-        if Sys.isbsd(p)
+        if Sys.isbsd(p) || sanitize(p) in ("memory", "memory_origins", "address") 
```

Comment:
Should we make a `prefer_clang(p)` helper rather than copy-pasting this everywhere?